sequencer/src/main/java/net/consensys/linea/metrics/HistogramMetrics.java

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package net.consensys.linea.metrics;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.function.DoubleSupplier;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.hyperledger.besu.plugin.services.MetricsSystem;
+import org.hyperledger.besu.plugin.services.metrics.Histogram;
+import org.hyperledger.besu.plugin.services.metrics.LabelledGauge;
+import org.hyperledger.besu.plugin.services.metrics.LabelledMetric;
+
+@Slf4j
+public class HistogramMetrics {
```

Comment:
Why can't we just reuse Micrometer's histogram here? [Here](https://github.com/Consensys/linea-monorepo/blob/main/jvm-libs/linea/metrics/micrometer/src/main/kotlin/net/consensys/linea/metrics/micrometer/MicrometerHistogramAdapter.kt) is an example how it's done and abstracted in Linea monorepo

Reply:
After examining Besu's source code I think I figured. You're not using Micrometer and you didn't use Histograms yet.

But maybe in order to be future proof, maybe we can instead add a histogram implementation into Besu directly and import it here? It's not that histograms are specific to Linea

Reply:
Indeed part of this feature is to add histogram support into Besu, for the moment there is a preview in the fork https://github.com/Consensys/linea-besu/compare/zkbesu...Consensys:linea-besu:introduce-summary-metrics, that will be ported upstream after we refine it